proxy/envoy/config.go

```diff
@@ -167,8 +167,17 @@ func buildHTTPListener(routeConfig *HTTPRouteConfig, ip string, port int, rds bo
 		config.RouteConfig = routeConfig
 	}
 
+	var sslContext *SSLContext
+	// Build the sslContext for http inbound listener (only inbound listener does not use rds)
+	if !rds {
+		sslContext = buildListenerSSLContext(mesh)
```

Comment:
+1. Please do not use nils. All functions should return actual values. I know that @myidpt did the opposite, but I let it slip to let that PR in faster.